langchain/vectorstores/pinecone.ts

```diff
@@ -0,0 +1,116 @@
+import type { PineconeClient } from "pinecone-client";
```

Comment:
We should make sure these imports don't fail if `pinecone-client` is not installed. We do this in a few other places as well for optional deps (e.g. see https://github.com/hwchase17/langchainjs/blob/80356338749618620cefdd1630120a570f5f1f62/langchain/llms/openai.ts#L18-L23)

Reply:
I think in this case it should be fine to keep it as-is, as we're passing the instance of the client as an argument, not creating the instance ourselves. 

The type import is removed by `tsc` during build phase and the resulting JS file does not import "pinecone-client" at all.

<img width="1283" alt="CleanShot 2023-02-19 at 02 57 26@2x" src="https://user-images.githubusercontent.com/1443449/219908012-689ba9de-454d-4ce2-b6fa-0a25b57b5ac2.png">

Reply:
It might be the case that we need to actually create the instance ourselves, possibly due to serialisation? @sullivan-sean

Reply:
oh yeah I see, this is fine!